apps/aecore/src/aec_genesis_block_settings.erl

```diff
@@ -0,0 +1,25 @@
+-module(aec_genesis_block_settings).
+
+-export([dir/0,
+         preset_accounts/0]).
+
+dir() ->
+    {ok, ProjectDir} = file:get_cwd(),
+    filename:join(ProjectDir, ".genesis").
+
+preset_accounts() ->
+    PresetAccountsFile = filename:join([dir(), "accounts.json"]),
+    case file:read_file(PresetAccountsFile) of
+        {error, _Err} ->
+            % no setup, no preset accounts
+            [];
+        {ok, JSONData} ->
+            Accounts =
+                lists:map(
+                    fun({EncodedPubKey, Amt}) ->
+                        {base64:decode(EncodedPubKey), Amt}
+                    end,
+                    jsx:decode(JSONData)),
+            % ensure deterministic ordering of accounts
+            lists:keysort(1, Accounts)
```

Comment:
This function is reading from file and sorting a potentially large list.
Maybe the result of this function - or the whole genesis block and block header - shall be cached in an ETS table lazily initialized?

Reply:
This may avoid the need for the `genesis_block_hash` change?